errors/errors.go

```diff
@@ -0,0 +1,220 @@
+// if the ErrorCodes is enabled
+// TODO: @robertojrojas update when feature is ready.
+func New(err error, metadata map[string]string, options ...Option) *Error {
+	// The following condition can be removed once the
+	// Error Codes Feature is GA
+	if !featureEnabled(metadata) {
+		return nil
+	}
+
+	if err == nil {
+		return nil
+	}
+
+	// Use default values
+	de := &Error{
+		err:            err,
+		reason:         unknown,
+		httpCode:       unknownHTTPCode,
+		grpcStatusCode: codes.Unknown,
+	}
+
+	// Now apply any requested options
+	// to override
+	for _, option := range options {
+		option(de)
+	}
+
+	return de
+}
+
+// Error implements the error interface.
+func (e *Error) Error() string {
+	if e != nil && e.err != nil {
+		return e.err.Error()
+	}
+	return ""
+}
+
+// Unwrap implements the error unwrapping interface.
+func (e *Error) Unwrap() error {
+	return e.err
+}
+
+// Description returns the description of the error.
+func (e *Error) Description() string {
+	if e.description != "" {
+		return e.description
+	}
+	return e.err.Error()
+}
+
+func WithErrorReason(reason string, httpCode int, grpcStatusCode codes.Code) Option {
+	f := func(er *Error) {
+		er.reason = reason
+		er.grpcStatusCode = grpcStatusCode
+		er.httpCode = httpCode
+	}
+	return f
+}
+
+func WithResourceInfo(resourceInfo *ResourceInfo) Option {
+	f := func(e *Error) {
+		e.resourceInfo = resourceInfo
+	}
+	return f
+}
+
+func WithDescription(description string) Option {
+	f := func(e *Error) {
+		e.description = description
+	}
+	return f
+}
+
+func WithMetadata(md map[string]string) Option {
+	f := func(e *Error) {
+		e.metadata = md
+	}
+	return f
+}
+
+func newErrorInfo(reason string, md map[string]string) *errdetails.ErrorInfo {
+	ei := errdetails.ErrorInfo{
+		Domain:   Domain,
+		Reason:   reason,
+		Metadata: md,
+	}
+
+	return &ei
+}
+
+func newResourceInfo(rid *ResourceInfo, err error) *errdetails.ResourceInfo {
+	return &errdetails.ResourceInfo{
+		ResourceType: rid.Type,
+		ResourceName: rid.Name,
+		Owner:        Owner,
+		Description:  err.Error(),
+	}
+}
+
+// *** GRPC Methods ***
+
+// GRPCStatus returns the gRPC status.Status object.
+func (e *Error) GRPCStatus() *status.Status {
+	messages := []protoiface.MessageV1{
+		newErrorInfo(e.reason, e.metadata),
+	}
+
+	if e.resourceInfo != nil {
+		messages = append(messages, newResourceInfo(e.resourceInfo, e.err))
+	}
+
+	ste, stErr := status.New(e.grpcStatusCode, e.description).WithDetails(messages...)
+	if stErr != nil {
+		return status.New(e.grpcStatusCode, e.description)
+	}
+
+	return ste
+}
+
+// *** HTTP Methods ***
+
+// ToHTTP transforms the supplied error into
+// a GRPC Status and then Marshals it to JSON.
+// It assumes if the supplied error is of type Error.
+// Otherwise, returns the original error.
+func (e *Error) ToHTTP() (int, []byte) {
+	if resp, sej := protojson.Marshal(e.GRPCStatus().Proto()); sej == nil {
```

Comment:
What is sej?

Reply:
I was using an `err` variable and I used the name to differentiate. I'm renaming to `err`.